src/Factory.php

```diff
@@ -37,20 +41,98 @@ public function __construct(array $config = [], Client $client = null, array $cl
     }
 
     /**
-     * Return an instance of a Endpoint based on the method called.
+     * Returns the Orange DAM API Endpoint requested by name.
+     *
+     *
+     */
+    public function getEndpoint(string $endpoint_name, mixed $args = []): Endpoint
+    {
+        $endpoint_class = 'Chromatic\\OrangeDam\\Endpoints\\' . $endpoint_name;
+        try {
+            $endpoint = new $endpoint_class($this->client, ...$args);
+        } catch (\Throwable $e) {
+            throw new OrangeDamUnimplementedEndpointException($endpoint_name, $e);
+        }
+
+        return $endpoint;
+    }
+
+    /**
+     * Returns an Orange Dam API AssetLink endpoint.
+     *
+     * Use getEndpoint('AssetLink') instead.
+     *
+     * @deprecated
+     */
+    public function assetLink(): AssetLink
+    {
+        return $this->getEndpoint('AssetLink');
+    }
+
+    /**
+     * Returns an Orange Dam API DataTable endpoint.
+     *
+     * Use getEndpoint('DataTable') instead.
+     *
+     * @deprecated
+     */
+    public function dataTable(): DataTable
+    {
+        return $this->getEndpoint('DataTable');
+    }
+
+    /**
+     * Returns an Orange Dam API MediaFile endpoint.
+     *
+     * Use getEndpoint('MediaFile') instead.
+     *
+     * @deprecated
+     */
+    public function mediaFile(): MediaFile
+    {
+        return $this->getEndpoint('MediaFile');
+    }
+
+    /**
+     * Returns an Orange Dam API OAuth2 endpoint.
+     *
+     * Use getEndpoint('OAuth2') instead.
+     *
+     * @deprecated
+     */
+    public function oAuth2(): OAuth2
+    {
+        return $this->getEndpoint('OAuth2');
+    }
+
+    /**
+     * Returns an Orange Dam API ObjectManagement endpoint.
+     *
+     * Use getEndpoint('ObjectManagement') instead.
+     *
+     * @deprecated
      */
-    public function __call(string $name, mixed $args): Endpoint
```

Comment:
@apotek Do you feel like we should leave the `__call` method but deprecate it? Removing this would break existing code right? 😞

Reply:
No. I have been testing this all morning. I implemented `assetLink()`, `dataTable()` etc so that we _don't_ need `__call()`. I provided declared methods for all the implemented Endpoints. I tested about twenty different migration commands using this, without changing the calling code.

Reply:
Oh, I see. Great!